engines/stark/ui/menu/settingsmenu.h

```diff
@@ -0,0 +1,194 @@
+/* ResidualVM - A 3D game interpreter
+ *
+ * ResidualVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the AUTHORS
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef STARK_UI_MENU_SETTINGS_MENU_H
+#define STARK_UI_MENU_SETTINGS_MENU_H
+
+#include "engines/stark/ui/menu/locationscreen.h"
+#include "engines/stark/services/services.h"
+#include "engines/stark/services/settings.h"
+#include "engines/advancedDetector.h"
+
+namespace Stark {
+
+class VisualImageXMG;
+
+/**
+ * Manager of test sound
+ */
+class SoundManager {
+public:
+	SoundManager();
+	~SoundManager() {}
+
+	/** Load sounds **/
+	void load();
+
+	/** Close the sound manager and reset pointers **/
+	void close();
+
+	/** play a specific sound in a loop */
+	void play(int index);
+
+	/** request to end the playing loop */
+	void endLoop();
+
+	/** stop any currently playing sound */
+	void stop();
+
+	/** update on game frame */
+	void update();
+
+private:
+	Resources::Sound *_currentSound;
+	Resources::Sound *_sounds[3];
+	bool _isLopping;
+};
+
+/**
+ * The setting menu of the game
+ */
+class SettingsMenuScreen : public StaticLocationScreen {
+public:
+	SettingsMenuScreen(Gfx::Driver *gfx, Cursor *cursor);
+	virtual ~SettingsMenuScreen();
+
+	// StaticLocationScreen API
+	void open() override;
+	void close() override;
+	void onMouseMove(const Common::Point &pos) override;
+
+	void handleMouseUp();
+
+private:
+	enum HelpTextIndex {
+		kHighRes = 5,
+		kSubtitles = 7,
+		kSpecialFX = 9,
+		kShadows = 11,
+		kHighResFMV = 13,
+		kVoice = 16,
+		kMusic = 18,
+		kSfx = 20,
+		kAllowFF = 22
+	};
+
+	enum WidgetIndex {
+		kWidgetVoice = 15,
+		kWidgetMusic = 17,
+		kWidgetSfx = 19
+	};
+
+	template<HelpTextIndex N>
+	void textHandler(StaticLocationWidget &widget, const Common::Point &mousePos);
+
+	template<Settings::BoolSettingIndex N>
+	void flipSettingHandler();
+
+	void backHandler();
+
+private:
+	static const uint32 _textColorHovered = 0xFF961E1E;
+	static const uint32 _textColorDefault = 0xFF000000;
+
+	SoundManager _soundManager;
+
+	bool isDemo() {
+		return StarkGameDescription->flags & ADGF_DEMO;
```

Comment:
There are now two implementations of `isDemo`. I think it'd be better to put one in StarkSettings and reuse it, now.